### PR TITLE
Makefile.include: add $(MAKEFILEDIR) helper and use it

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The `$(abspath $(dir $(lastword $(MAKEFILE_LIST))))` is used to get the location of the current Makefile.

This is a useful helper especially for external modules.
It is used in several places in RIOT already, so add a special variable for it to make it more accessable.

### Testing procedure

Murdock should still be able to build everything.

### CODE REVIEW

**Comment by miri64** [comment]
Why not use the name `MAKEFILEDIR`? `THISDIR` sounds confusing and it isn't clear what "this" is refering to, especially considering there is also `CURDIR`. Granted, `MAKEFILEDIR` can also be misunderstood, but I think its a bit clearer than `THISDIR`.

File: tests/cpp_exclude/module_exclude/Makefile.include [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Use an immediate variable to evaluate `MAKEFILE_LIST` now
```
Comment (miri64) [inline tests/cpp_exclude/module_exclude/Makefile.include:1]: I guess comments of this type should be adopted then ;-).

**Comment by miri64** [comment]
Please squash and change the commit messages in accordance with the current state.

**Comment by benpicco** [comment]
squashed.

**Comment by miri64** [comment]
@fjmolinas @aabadie could one of you have a look if this is okay like this?

File: Makefile.include [hunk 2]
```diff
@@ -8,8 +8,11 @@ ifeq (,$(filter $(MATCH_MAKE_VERSION),$(MAKE_VERSION)))
       $(MATCH_MAKE_VERSION) or later.)
 endif
 
+# will evaluate to the absolute path of the Makefile it's evaluated in
+MAKEFILEDIR = $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+
 # 'Makefile.include' directory, must be evaluated before other 'include'
-_riotbase := $(dir $(lastword $(MAKEFILE_LIST)))
+_riotbase := $(MAKEFILEDIR)
```
Comment (fjmolinas) [inline Makefile.include:15]: This is changing `RIOTBASE` to use absolute path instead of relative paths, not sure if this can have an impact..

Comment (benpicco) [reply Makefile.include:15]: I can drop this one - or make `MAKEFILEDIR` use relative paths.

From my gut feeling absolute paths should be more 'stable' though.

Comment (fjmolinas) [reply Makefile.include:15]: Never-mind its overridden to an abspath anyway:

```
override RIOTBASE       := $(abspath $(RIOTBASE))
```

**Comment by fjmolinas** [comment]
I'm found an issue with this PR, which could deserve a revert unless fixed, I know realized that the name and the behaviour is error prone (thanks @cladmi for the catch), `MAKEFILEDIR` does not hold the name of the current Makefile, but the last included Makefile, so the variable will change if another `makefile` is included, e.g.:

https://github.com/RIOT-OS/RIOT/blob/bb1ff1f6356c026b308789d27a82285532b350c2/Makefile.include#L11-L20

after L20 `MAKEFILEDIR` will hold the value of the last `Makefile` in `RIOT_MAKEFILES_GLOBAL_PRE` so if used afterwards it will hold a deceiving value. It would need to be immediately expanded on every usage, and have a better defined name "`dir_of_last_included_makefile`", something that would force people to read the doc or force immediate expansion.

IMO if this is still wanted it should be re-implemented, and its simpler to revert for now.

**Comment by fjmolinas** [comment]
> IMO if this is still wanted it should be re-implemented, and its simpler to revert for now.
> I'm found an issue with this PR, which could deserve a revert unless fixed, I know realized that the name and the behaviour is error prone (thanks @cladmi for the catch), `MAKEFILEDIR` does not hold the name of the current Makefile, but the last included Makefile, so the variable will change if another `makefile` is included, e.g.:
> 
> https://github.com/RIOT-OS/RIOT/blob/bb1ff1f6356c026b308789d27a82285532b350c2/Makefile.include#L11-L20
> 
> after L20 `MAKEFILEDIR` will hold the value of the last `Makefile` in `RIOT_MAKEFILES_GLOBAL_PRE` so if used afterwards it will hold a deceiving value. It would need to be immediately expanded on every usage, and have a better defined name "`dir_of_last_included_makefile`", something that would force people to read the doc or force immediate expansion.
> 
> IMO if this is still wanted it should be re-implemented, and its simpler to revert for now.
> 

Note that in all the current uses it would not lead to issues since its assigned with `:=` or used only once. But a name change, and more doc is at least needed, to ensure people are aware of the behaviour when using it.

**Pull request merged**